src/ansys/sound/core/sound_composer/source_control_time.py

```diff
@@ -0,0 +1,140 @@
+from ansys.dpf.core import Field, Operator
+from matplotlib import pyplot as plt
+
+from ansys.sound.core.signal_utilities.load_wav import LoadWav
+
+from .._pyansys_sound import PyAnsysSoundException
+from ._source_control_parent import SourceControlParent
+
+ID_LOAD_FROM_TEXT = "load_sound_samples_from_txt"
+
+
+class SourceControlTime(SourceControlParent):
+    """Sound Composer's spectrum source's control class.
+
+    This class stores the control parameters used by the Sound Composer for a BBN or harmonic
+    source.
+    """
+
+    def __init__(self, file_str: str = ""):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file_str : str, default: ""
+            Path to the control data file (WAV or text format).
+        """
+        super().__init__()
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_LOAD_FROM_TEXT)
+
+        if len(file_str) > 0:
+            if file_str.endswith(".wav"):
+                self.load_from_wave_file(file_str)
+            else:
+                self.load_from_text_file(file_str)
+        else:
+            self.control = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.control is None:
+            return "Not set"
+        else:
+            support_data = self.control.time_freq_support.time_frequencies.data
+            str_duration = f"{support_data[-1]:.1f} s" if len(support_data) > 0 else "N/A"
+            return (
+                f"Unit: {self.control.unit}\n"
+                f"Duration: {str_duration}\n"
+                f"Min - max: {self.control.data.min():.1f} - {self.control.data.max():.1f} "
+                f"{self.control.unit}"
+            )
+
+    @property
+    def control(self) -> Field:
+        """Control profile (control parameter values over time). Unit depends on loaded data."""
+        return self.__control
+
+    @control.setter
+    def control(self, control: Field):
+        """Set the control."""
+        if control is not None:
+            if not (isinstance(control, Field) or control is None):
+                raise PyAnsysSoundException(
+                    "Specified control profile must be provided as a DPF field."
+                )
+
+            if len(control.data) == 0:
+                raise PyAnsysSoundException(
+                    "Specified control profile must have at least one element."
+                )
+
+        self.__control = control
+
+    def load_from_wave_file(self, file_str: str):
+        """Load control data from a WAV file.
+
+        Parameters
+        ----------
+        file_str : str
+            Path to the WAV file.
+        """
+        loader = LoadWav(file_str)
+        loader.process()
+        self.control = loader.get_output()[0]
+
+    def load_from_text_file(self, file_str: str):
+        """Load control data from a text file.
```

Comment:
@ansaminard it would be good to give some information about the expected content of the text file here

Reply:
done